src/container.jl

```diff
@@ -1,81 +1,79 @@
-mutable struct Trace{Tspl<:AbstractSampler, Tvi<:AbstractVarInfo, Tmodel<:Model}
-    model::Tmodel
-    spl::Tspl
-    vi::Tvi
-    ctask::CTask
-
-    function Trace{SampleFromPrior}(model::Model, spl::AbstractSampler, vi::AbstractVarInfo)
-        return new{SampleFromPrior,typeof(vi),typeof(model)}(model, SampleFromPrior(), vi)
-    end
-    function Trace{S}(model::Model, spl::S, vi::AbstractVarInfo) where S<:Sampler
-        return new{S,typeof(vi),typeof(model)}(model, spl, vi)
-    end
+struct Trace{F}
+    f::F
```

Comment:
> Maybe consider making TracedModel and TracedFun as subtypes of AbstractMCMC.AbstractModel? It slightly improves clarity, and also prevent potential misuses.

Yes, maybe that could be done. On the other hand, such type constraints can be quite restrictive and it didn't seem necessary to introduce them at this stage. Also maybe even `FixedModel` would be more appropriate since `TracedModel` wraps a `Model` and a specific sampler and `VarInfo` object (I guess it should also include the RNG but this is not part of the current implementation and hence I did not want to change it in these PRs).